File: backend/src/main/java/pt/isel/rugby/model/AthleteInjury.java
package pt.isel.rugby.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Table(name = "athlete_injury")
@Data
public class AthleteInjury implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne
    private Injury injury;

    @ManyToOne
    private Athlete athlete;

    @Column
    private String state;

    @Column
    private Date beginDate;

    @Column
    private Date returnDate;

    @Column
    private String recommendations;
}

File: frontend2/Rugby/src/app/classes/associations/AthleteInjury.ts
import {Opponent} from '../opponent';
import {Athlete} from '../athlete';
import { Injury } from '../injury';

export class AthleteInjury {
  constructor(
    public id?: number,
    public severity?: string,
    public state?: string,
    public beginDate?: Date,
    public returnDate?: Date,
    public recommendations?: string,
    public athlete?: Athlete,
    public injury?: Injury
  ) {
    this.id = id ? id : 0;
    this.severity = severity ? severity : '';
    this.state = state ? state : '';
    this.beginDate = beginDate ? beginDate : new Date(0);
    this.returnDate = returnDate ? returnDate : new Date(0);
    this.recommendations = recommendations ? recommendations : '';
    this.athlete = athlete ? athlete : new Athlete();
    this.injury = injury ? injury : new Injury();
  }
}
